modules/drivers/mongo/src/metabase/driver/mongo/query_processor.clj

```diff
@@ -696,22 +699,15 @@
     {$sum {$cond {:if   (->rvalue arg)
                   :then 1
                   :else 0}}}
-    [:avg arg]
-    {$avg (->rvalue arg)}
 
+    ;; these aggregation types can all be used in expressions as well so their implementations live above in the
+    ;; general [[->rvalue]] implementations
+    #{:avg :stddev :sum :min :max}
+    (->rvalue &match)
```

Comment:
Took advantage of the `mbql/match` support for multiple clauses in one pattern to eliminate a bunch of repetitive code -- all of these aggregation types just handed off to `->rvalue`.